portality/scripts/link_checker_report.py

```diff
@@ -38,26 +46,84 @@ def write_results(df, filename='multi_result.csv'):
     print("Result CSV file has been written.")
 
 
-def fetch_matching_rows(df, report_values):
+def _get_link_type(link, journal):
+    form = JournalFormXWalk.obj2form(journal)
```

Comment:
This is quite a lot of work - we're iterating through the object twice - the Xwalk to the form, then the properties of the form.

Reply:
It's the only way to do it, I'm afraid, as the result needs to connect the user to the form.  Also, iterating through the form is easier than iterating through the model.

It seems like a lot of work, but in reality I have brought the run time of this script down from 12 hours to 2 minutes, so I think it's ok